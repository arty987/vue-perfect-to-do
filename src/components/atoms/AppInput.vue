<script>
export default {
  name: "AppInput",
  data() {
    return {
      inputValue: "",
    };
  },
  methods: {
    addTask() {
      const inputValue = this.inputValue.trim();
      if (inputValue) {
        this.$store.dispatch("addTask", inputValue);
        this.inputValue = "";
      }
    },
    clearInputValue() {
      this.inputValue = "";
    }
  }
}
</script>

<template>
  <input
      type="text"
      placeholder="Please type your task here..."
      v-model="inputValue"
      maxlength="150"
      @keyup.enter="addTask"
      @keyup.esc="clearInputValue"
  />
</template>

<style scoped>
input {
  width: 100%;
  margin-left: 10px;
  outline: none;
  border: none;
}
</style>