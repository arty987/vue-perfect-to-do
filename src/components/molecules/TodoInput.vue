<script>
import AppCheckbox from "@/components/atoms/AppCheckbox.vue";
import AppInput from "@/components/atoms/AppInput.vue";

export default {
  name: "TodoInput",
  components: {
    AppInput,
    AppCheckbox
  },
  data() {
    return {
      inputValue: "",
    };
  },
  methods: {
    addTask() {
      const inputValue = this.inputValue.trim();
      if (inputValue) {
        this.$store.dispatch("addTask", inputValue);
        this.inputValue = "";
      }
    },
    clearInputValue() {
      this.inputValue = "";
    },
  }
}
</script>

<template>
  <li class="list-group-item d-flex justify-content-start align-items-center">
    <AppCheckbox :disabled="true"/>
    <AppInput placeholder="Please type your task here..."
              v-model="inputValue"
              @enter="addTask"
              @esc="clearInputValue"/>
  </li>
</template>

<style scoped lang="scss">
.list-group-item {
  margin-bottom: 20px;
  border-bottom-right-radius: var(--bs-border-radius);
  border-bottom-left-radius: var(--bs-border-radius);
}

.app-input {
  width: 100%;
  margin-left: 10px;
  outline: none;
  border: none;
}
</style>