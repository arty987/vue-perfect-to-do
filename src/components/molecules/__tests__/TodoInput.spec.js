import {shallowMount} from '@vue/test-utils'
import TodoInput from "@/components/molecules/TodoInput.vue";
import {createStore} from "vuex";
import AppInput from "@/components/atoms/AppInput.vue";

describe('TodoInput.vue', () => {

    let actions;
    let store;

    beforeEach(() => {
        actions = {
            addTask: jest.fn()
        };

        store = createStore({
            actions
        });
    });

    it('Should render TodoInput', () => {
        const wrapper = shallowMount(TodoInput);
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Should add new task on enter', async () => {
        const wrapper = shallowMount(TodoInput, {
            global: {
                plugins: [store]
            }
        });

        const input = wrapper.findComponent(AppInput);

        await input.setValue('new task');
        await input.trigger('enter');

        expect(actions.addTask).toHaveBeenCalledWith(expect.any(Object), 'new task');
        expect(input.props().modelValue).toBe('');
    });

    it('Should clear input field on esc', async () => {
        const wrapper = shallowMount(TodoInput, {
            global: {
                plugins: [store]
            }
        });

        const input = wrapper.findComponent(AppInput);

        await input.setValue('new task');
        await input.trigger('esc');

        expect(actions.addTask).not.toHaveBeenCalled();
        expect(input.props().modelValue).toBe('');
    });
})