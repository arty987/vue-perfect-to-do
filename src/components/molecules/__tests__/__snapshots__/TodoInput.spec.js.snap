// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput.vue Should render TodoInput 1`] = `
<li
  class="list-group-item d-flex justify-content-start align-items-center"
>
  <app-checkbox-stub
    checked="false"
    disabled="true"
  />
  <app-input-stub
    modelvalue=""
    placeholder="Please type your task here..."
  />
</li>
`;
